### PR TITLE
Add gen RO5 and move PopularTags to heap

### PR DESCRIPTION
**Pull request by Hroy02**

### CODE REVIEW

File: src/User.cpp [hunk 1]
```diff
@@ -100,7 +100,57 @@ Watchable* RerunRecommenderUser::getRecommendation(Session& s) {
...
+        this->cleanPopularTags(true);
+        this->deepCopyPopularTags(rhs._popularTagsVector);
+    }
+    return *this;
+}
+
+GenreRecommenderUser::GenreRecommenderUser(GenreRecommenderUser&& rhs) : User(rhs),
+                                                                _popularTagsMap(rhs._popularTagsMap),
+                                                                _popularTagsVector(rhs._popularTagsVector) {
+    rhs.cleanPopularTags(false);
```
Comment (IdanTo1) [inline src/User.cpp]: is it needed? isn't it happening thanks to vector::clear?

File: src/User.cpp [hunk 2]
```diff
@@ -100,7 +100,57 @@ Watchable* RerunRecommenderUser::getRecommendation(Session& s) {
...
+}
+
+GenreRecommenderUser& GenreRecommenderUser::operator=(GenreRecommenderUser&& rhs) {
+    if(this != &rhs) {
+        this->cleanPopularTags(true);
+        _popularTagsMap = rhs._popularTagsMap;
+        _popularTagsVector = rhs._popularTagsVector;
+        rhs.cleanPopularTags(false);
```
Comment (IdanTo1) [inline src/User.cpp:148]: same

File: src/User.cpp [hunk 3]
```diff
@@ -112,14 +162,14 @@ void GenreRecommenderUser::updatePopularTags() {
...
-            _popularTagsMap[*iter] = &(_popularTagsVector.back());
+            _popularTagsVector.push_back(new PopularTag(*iter));
+            _popularTagsMap[*iter] = _popularTagsVector.back();
         }
         else {
             _popularTagsMap[*iter]->increaseCount();
         }
         // sorts by PopularTag's operator<()
-        std::sort(_popularTagsVector.begin(), _popularTagsVector.end());
+        std::sort(_popularTagsVector.begin(), _popularTagsVector.end(), [](PopularTag* a, PopularTag* b) { return *a < *b; });
```
Comment (IdanTo1) [inline src/User.cpp]: line is too long
great implementation!

**Review by IdanTo1 — CHANGES_REQUESTED** [review]

File: src/User.cpp (hunk 1 above)
Comment (Hroy02) [reply src/User.cpp]: Very needed, otherwise the destructor will deep clean

**Review by IdanTo1 — APPROVED** [review]

**Pull request merged**